### PR TITLE
Add function GetFileContent to Google Drive

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
resolves https://github.com/wso2-enterprise/choreo/issues/6413

### CODE REVIEW

File: drive/utils.bal [hunk 1]
```diff
@@ -662,3 +665,21 @@ isolated function uploadFileUsingByteArray(http:Client httpClient, byte[] byteAr
...
 }
 
+# Utility function to retrieve File content as a binary content with its MIME type.
+# 
+# + httpClient - Drive client
+# + path - GET URI path
+# + return - `FileContent` if successful, `error` if not.
+isolated function generateRecordFileContent(http:Client httpClient, string path) returns @tainted FileContent|error {
+    http:Response response = <http:Response> check httpClient->get(<@untainted>path);
+    if (response.statusCode is http:STATUS_OK) {
```
Comment (SanduDS) [inline drive/utils.bal:675]: Usually, we get status code as int and we use "is" to check the types like string, null, int etc. So Isn't it should be  "==" ?

File: drive/utils.bal [hunk 2]
```diff
@@ -662,3 +665,21 @@ isolated function uploadFileUsingByteArray(http:Client httpClient, byte[] byteAr
...
+        FileContent fileContent  = {
+            content: check response.getBinaryPayload(),
+            mimeType: response.getContentType()
+        }; 
+        return fileContent;
+    } else {
+        return error(ERR_EXTRACTING_BINARY_PAYLOAD);
```
Comment (SanduDS) [inline drive/utils.bal:682]: Just a suggestion, We can use `fail error(<Error Messsage>)`;

File: drive/utils.bal [hunk 3]
```diff
@@ -662,3 +665,21 @@ isolated function uploadFileUsingByteArray(http:Client httpClient, byte[] byteAr
...
 
+# Utility function to retrieve File content as a binary content with its MIME type.
+# 
+# + httpClient - Drive client
+# + path - GET URI path
+# + return - `FileContent` if successful, `error` if not.
+isolated function generateRecordFileContent(http:Client httpClient, string path) returns @tainted FileContent|error {
+    http:Response response = <http:Response> check httpClient->get(<@untainted>path);
+    if (response.statusCode is http:STATUS_OK) {
+        FileContent fileContent  = {
```
Comment (SanduDS) [inline drive/utils.bal:676]:
```suggestion
        return {
            content: check response.getBinaryPayload(),
            mimeType: response.getContentType()
        }; 
```
Does this suggestion reduce the code readability?

File: drive/utils.bal (hunk 1 above)
Comment (SanduDS) [reply drive/utils.bal:675]: I checked that both work fine. So what can be the best? any other ideas:)?

File: drive/utils.bal (hunk 3 above)
Comment (SkNuwanTissera) [reply drive/utils.bal:676]: Yeah. This can improve readability. Let's change it.

Comment (SkNuwanTissera) [reply drive/utils.bal]: Changed it /Thanks

File: drive/utils.bal (hunk 2 above)
Comment (SkNuwanTissera) [reply drive/utils.bal]: Yeah. That is better. I changed it. /Thanks

File: drive/utils.bal (hunk 1 above)
Comment (SkNuwanTissera) [reply drive/utils.bal]: yeah. That is good. I changed it
/Thanks

**Review by abeykoon — APPROVED** [review]

**Pull request merged**